xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-ui/src/main/resources/AnnotationCode/Script.xml

```diff
@@ -560,70 +560,90 @@ return XWiki;
       <cache>long</cache>
     </property>
     <property>
-      <code>## Retrieve the annotation settings from the configuration object
-#set($config = 'AnnotationCode.AnnotationConfig')
-#set($configObj = $xwiki.getDocument($config).getObject($config))
-#set($annotationHighlightByDefault = $configObj.getProperty('displayHighlight').value)
-#set($annotationsDisplayedByDefault = $configObj.getProperty('displayed').value)
-#set($annotationsActivated = $configObj.getProperty('activated').value)
-#set($exceptionSpaces = $configObj.getProperty('exceptionSpaces').value)
-#set($annotationClass = $configObj.getProperty('annotationClass').value)
+      <code>// Retrieve the annotation settings from the configuration object
+const config = JSON.parse(document.getElementById('annotation-config').text);
+
+// TODO: Many messages below rely on string concatenation instead of parameters (XWIKI-21961).
+define('xwiki-annotation-messages', {
+  prefix: '',
+  keys: [
+    'annotations.annotated.error.noannotatedelement',
+    'annotations.annotated.error.wrongsyntax',
+    'annotations.menu.loading',
+    'annotations.menu.loaderror',
+    'annotations.annotated.loading',
+    'annotations.annotated.loaderror',
+    'annotations.annotated.loaderror.wrongresponse',
+    'annotations.action.delete.confirm',
+    'annotations.action.delete.inProgress',
+    'annotations.action.delete.done',
+    'annotations.action.delete.failed',
+    'annotations.action.validate.success',
+    'annotations.action.validate.loaderror',
+    'annotations.action.edit.form.loaderror',
+    'annotations.action.view.form.loaderror',
+    'annotations.action.edit.success',
+    'annotations.action.edit.loaderror',
+    'annotations.action.create.error.wrongsyntax',
+    'annotations.action.create.selection.invalid',
+    'annotations.action.create.form.loaderror',
+    'annotations.action.create.success',
+    'annotations.action.create.loaderror'
+  ]
+})
 
-var XWiki = (function (XWiki) {
+require(['xwiki-l10n!xwiki-annotation-messages'], function(l10n) {
+  XWiki = (function (XWiki) {
```

Comment:
Why keep the anonymous function? Its purpose was to create a namespace so that we don't pollute the global namespace, but the require callback function does this now.

Reply:
But don't we still want to expose `XWiki.Annotation` publicly?

Reply:
ok, now I get what you mean, indeed I can drop a level of wrapping. Thanks!